src/session/session.c

```diff
@@ -2741,8 +2740,13 @@ session_set_conn_entry_data (THREAD_ENTRY * thread_p, SESSION_STATE * session_p)
   if (thread_p->conn_entry != NULL)
     {
       /* If we have a connection entry associated with this thread, setup session data for this conn_entry */
-      thread_p->conn_entry->session_p = session_p;
-      thread_p->conn_entry->session_id = session_p->id;
+      thread_p->conn_entry->set_session (session_p);
```

Comment:
thread_p->set_session (..)
thread_p->set_session_id (..)
and remove set_connectionless...